### PR TITLE
ddl: arg v2 for flashback cluster and alter table attributes

### PR DESCRIPTION
**Pull request by joechenrh**
### What problem does this PR solve?

Issue Number: Ref #53930 #54436

Problem Summary:

### What changed and how does it work?

As title.

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/meta/model/job_args.go [hunk 1]
```diff
@@ -1291,6 +1291,18 @@ func GetAddCheckConstraintArgs(job *Job) (*AddCheckConstraintArgs, error) {
...
+// KeyRange is copied from kv.KeyRange to avoid cycle import.
+type KeyRange struct {
+	StartKey []byte `json:"start_key,omitempty"`
+	EndKey   []byte `json:"end_key,omitempty"`
+
+	// Unused fields
+	// They are reserved to ensure the consistent result after marshalling.
+	UnusedStruct struct{} `json:"-"`
+	UnusedSlice  []byte   `json:"-"`
+	UnusedInt    int32    `json:"-"`
```
Comment (D3Hunter) [inline pkg/meta/model/job_args.go:1303]:
```suggestion
```
we don't need them

File: pkg/meta/model/job_args.go [hunk 2]
```diff
@@ -1291,6 +1291,18 @@ func GetAddCheckConstraintArgs(job *Job) (*AddCheckConstraintArgs, error) {
 	return getOrDecodeArgsV2[*AddCheckConstraintArgs](job)
 }
 
+// KeyRange is copied from kv.KeyRange to avoid cycle import.
+type KeyRange struct {
+	StartKey []byte `json:"start_key,omitempty"`
+	EndKey   []byte `json:"end_key,omitempty"`
```
Comment (D3Hunter) [inline pkg/meta/model/job_args.go:1297]: keep json tag the same to be compatible

File: pkg/meta/model/job_args.go [hunk 3]
```diff
@@ -1366,6 +1378,35 @@ func (a *RecoverArgs) fillJob(job *Job) {
 	job.Args = []any{a}
 }
 
+// AlterTableAttributesArgs is the argument for alter table attributes
+type AlterTableAttributesArgs struct {
+	Rule *pdhttp.LabelRule `json:"rule,omitempty"`
```
Comment (D3Hunter) [inline pkg/meta/model/job_args.go:1383]:
```suggestion
	LabelRule *pdhttp.LabelRule `json:"label_rule,omitempty"`
```

File: pkg/meta/model/job_args.go [hunk 4]
```diff
@@ -1451,3 +1492,66 @@ func GetPlacementPolicyArgs(job *Job) (*PlacementPolicyArgs, error) {
...
+		}
+		if !a.EnableAutoAnalyze {
+			job.Args[3] = "OFF"
+		}
+		if !a.SuperReadOnly {
+			job.Args[4] = "OFF"
+		}
+		if !a.EnableTTLJob {
+			job.Args[8] = "OFF"
+		}
```
Comment (D3Hunter) [inline pkg/meta/model/job_args.go:1524]: prefer we name and value for args 3/4/8, and move this above `job.Args = []any`

File: pkg/meta/model/job_args.go [hunk 5]
```diff
@@ -1451,3 +1492,66 @@ func GetPlacementPolicyArgs(job *Job) (*PlacementPolicyArgs, error) {
...
+
+// FlashbackClusterArgs is the argument for flashback cluster.
+type FlashbackClusterArgs struct {
+	FlashbackTS        uint64         `json:"flashback_ts,omitempty"`
+	PDScheduleValue    map[string]any `json:"pd_schedule_value,omitempty"`
+	EnableGC           bool           `json:"enable_gc,omitempty"`
+	EnableAutoAnalyze  bool           `json:"enable_auto_analyze,omitempty"`
+	EnableTTLJob       bool           `json:"enable_ttl_job,omitempty"`
+	SuperReadOnly      bool           `json:"super_read_only,omitempty"`
+	LockedRegions      uint64         `json:"locked_regions,omitempty"`
```
Comment (D3Hunter) [inline pkg/meta/model/job_args.go:1504]:
```suggestion
	LockedRegionCnt      uint64         `json:"locked_region_cnt,omitempty"`
```

File: pkg/meta/model/job_test.go [hunk 6]
```diff
@@ -293,7 +293,6 @@ func TestJobCodec(t *testing.T) {
 		ID:         16,
 		Type:       ActionFlashbackCluster,
 		BinlogInfo: &HistoryInfo{},
-		Args:       []any{0, map[string]any{}, "ON", true},
```
Comment (D3Hunter) [inline pkg/meta/model/job_test.go:296]: does it matter if kept?

Comment (joechenrh) [reply pkg/meta/model/job_test.go:296]: The order of arguments here is wrong, and they're not used. So I just removed them.

File: pkg/meta/model/job_args.go (hunk 2 above)
Comment (joechenrh) [reply pkg/meta/model/job_args.go:1297]: Remove the json tags as `kv.KeyRange` does?

Comment (D3Hunter) [reply pkg/meta/model/job_args.go:1297]: yes

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Review by D3Hunter — APPROVED** [review]

File: pkg/meta/model/job_args_test.go [hunk 7]
```diff
@@ -829,3 +829,58 @@ func TestPlacementPolicyArgs(t *testing.T) {
...
+		require.Equal(t, inArgs.StartTS, args.StartTS)
+		require.Equal(t, inArgs.CommitTS, args.CommitTS)
+		require.Equal(t, inArgs.LockedRegionCnt, args.LockedRegionCnt)
+		require.Equal(t, inArgs.FlashbackKeyRanges, args.FlashbackKeyRanges)
+		require.Equal(t, inArgs.PDScheduleValue, args.PDScheduleValue)
+
+		require.True(t, args.EnableAutoAnalyze)
+		require.True(t, args.EnableGC)
+		require.True(t, args.EnableTTLJob)
+		require.True(t, args.SuperReadOnly)
```
Comment (joccau) [inline pkg/meta/model/job_args_test.go:865]: try require.Equal(t, inArgs, args)

File: pkg/ddl/executor.go [hunk 8]
```diff
@@ -5668,18 +5666,20 @@ func (e *executor) AlterTableAttributes(ctx sessionctx.Context, ident ast.Ident,
...
 		TableName:      meta.Name.L,
 		Type:           model.ActionAlterTableAttributes,
 		BinlogInfo:     &model.HistoryInfo{},
-		Args:           []any{rule},
 		CDCWriteSource: ctx.GetSessionVars().CDCWriteSource,
 		SQLMode:        ctx.GetSessionVars().SQLMode,
 	}
 
-	err = e.DoDDLJob(ctx, job)
+	pdLabelRule := pdhttp.LabelRule(*rule)
```
Comment (joccau) [inline pkg/ddl/executor.go:5680]: pdLabelRule := （*pdhttp.LabelRule)(rule)

**Review by joccau — COMMENTED** [review]
rest lgtm

**Review by joccau — APPROVED** [review]
LGTM

**Pull request merged**